plonky2/src/gates/poseidon.rs

```diff
@@ -589,24 +597,29 @@ mod tests {
             .collect::<Vec<_>>();
 
         let mut inputs = PartialWitness::new();
-        inputs.set_wire(
-            Wire {
-                row,
-                column: Gate::WIRE_SWAP,
-            },
-            F::ZERO,
-        );
-        for i in 0..SPONGE_WIDTH {
-            inputs.set_wire(
+        inputs
+            .set_wire(
                 Wire {
                     row,
-                    column: Gate::wire_input(i),
+                    column: Gate::WIRE_SWAP,
                 },
-                permutation_inputs[i],
-            );
+                F::ZERO,
+            )
+            .unwrap();
```

Comment:
since `IoResult` is also a `Result`, can you map the error instead of panicking?

Reply:
it's a unit test, and it doesn't return anything hence why I kept `unwrap()` (there are some pre-existing ones below)